ompi/mca/osc/ucx/osc_ucx_component.c

```diff
@@ -850,22 +850,113 @@ int ompi_osc_find_attached_region_position(ompi_osc_dynamic_win_info_t *dynamic_
                                                       base, len, insert);
     }
 }
+extern bool ompi_osc_need_acc_lock(ompi_osc_ucx_module_t *module, int target)
```

Comment:
No need for `extern` here and on the functions below since it has been declared `extern inline` already

Reply:
Not ready. @devreal Actually, looks like the new change (using ACC lock, instead of the main application lock) leads to hang at a large scale. The reason is that we must not perform window attach, while other processes are accessing this dynamic window. Currently, ACC lock only remains alive during the get_dynamic_window_info() which is not enough. We must lock ACC-lock during entire RMA operation. Working on this change.

Reply:
I'm just thinking out loud. Since we only need lock to read the window info (not the data inside the window), Ideally, we should not need to have the acc-lock during entire rma operation, but somehow, using this extra acc lock instead of app lock is leading to race condition.

Reply:
> The reason is that we must not perform window attach, while other processes are accessing this dynamic window.

I'm not sure I understand the issue. Why not? We shouldn't keep the acc longer than we need to, i.e., we should release it as soon as we have queried the attached segments.

Reply:
I need to investigate further to know exactly what's wrong with the current state of the patch that leads to hang. I don't get the hang if I omit the today's changes (adding a c-lock). I will update you.

Reply:
hey, @devreal I fixed the hang issue, please let me know if you have any more comments. Thanks.